src/renderer/store/handlers/channel.ts

```diff
@@ -277,10 +306,28 @@ const sendOnEnter = (event, resetTab) => async (dispatch, getState) => {
       )
 
       const identityAddress = identitySelectors.address(getState())
-      if (useTor && users[channel.id] && users[channel.id].onionAddress) {
+      const contact2 = contactsSelectors.contact(id)(getState())
+      const contactsListMessages2 = Object.values(contact2.messages)
+      let channelOnionAddress = null
+      let user = null
+      let onionAddress = null
+
+      if (contactsListMessages2[0]) {
+        channelOnionAddress = contactsListMessages2[0].onionAddress
```

Comment:
Instead, when receiving message, check if we have user in our contact info, and if not, add him there.